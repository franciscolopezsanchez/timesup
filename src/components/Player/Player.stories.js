import React from 'react';
import Player from './';
import { withKnobs, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

export default { title: 'Player', decorators: [withKnobs]  };

export const WithoutProps = () => <Player />;
export const WithName = () => <Player name={text('Name', 'Pacoli')} />;
export const WithNameAndOnDelete = () => <Player name={text('Name', 'Pacoli')} onRemovePlayer={action('Remove Player')}  />;
export const Width400 = () => <div style={{ width: '400px' }}>
    <Player name={text('Name', 'Pacoli')} onRemovePlayer={action('Remove Player')}  />
</div>;
export const Width200 = () => <div style={{ width: '200px' }}>
    <Player name={text('Name', 'Pacoli')} onRemovePlayer={action('Remove Player')}  />
</div>;

