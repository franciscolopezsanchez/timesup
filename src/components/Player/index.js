import React from "react"
import styles from "./Player.module.css"
import PropTypes from "prop-types"

function Player({name = "Unknown", onRemovePlayer}) {
  return (
    <div className={styles.row}>
      <img
        className={styles.avatar}
        src={`https://api.adorable.io/avatars/50/${name}@adorable.png`}
        alt={name}
      />
      <span className={styles.name}>{name}</span>
      {onRemovePlayer && <button onClick={onRemovePlayer}>X</button>}
    </div>
  )
}

Player.propTypes = {
  name: PropTypes.string,
  onRemovePlayer: PropTypes.func,
}

export default Player
