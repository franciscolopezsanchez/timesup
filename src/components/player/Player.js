import React from "react"

function Player({ name, removePlayer }) {
	if (!name) return null
	return (
		<div className="player">
			<span>{name}</span>
			<button onClick={() => removePlayer(name)}>X</button>
		</div>
	)
}

export default Player
