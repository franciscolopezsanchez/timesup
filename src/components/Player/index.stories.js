import React from 'react';
import Player from './Player';
import { withKnobs, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

export default { title: 'Player', decorators: [withKnobs]  };

export const DefaultPlayer = () => <Player name={text('Name', 'Pacoli')} removePlayer={action('Remove Player')} />;