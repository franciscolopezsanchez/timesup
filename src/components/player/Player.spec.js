import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react"
import Player from "./Player"

afterEach(cleanup)

describe("Player tests", () => {
	test("Player handles undefined name as argument", () => {
		// Arrange
		const name = undefined
		// Act
		const { container } = render(<Player name={name} />)
		// Assert
		expect(container.firstChild).toBeFalsy()
	})

	test("Player handles empty name as argument", () => {
		// Arrange
		const name = ""
		// Act
		const { container } = render(<Player name={name} />)
		// Assert
		expect(container.firstChild).toBeFalsy()
	})

	test("Player handles a name as argument", () => {
		// Arrange
		const name = "Jesus"
		// Act
		const { container, getByText } = render(<Player name={name} />)
		// Assert
		expect(container.firstChild).toBeTruthy()
		expect(getByText("Jesus")).toBeTruthy()
	})

	test("Player handles click on remove button", () => {
		// Arrange
		const name = "Jesus"
		const removePlayer = jest.fn()
		// Act
		render(<Player name={name} removePlayer={removePlayer} />)
		const button = document.querySelector("button")
		fireEvent.click(button)

		// Assert
		expect(removePlayer.mock.calls.length).toBe(1)
	})

	test("renders snapshot", () => {
		// Arrange

		// Act
		const { asFragment } = render(<Player name="Jesus" />)
		//Assert
		expect(asFragment()).toMatchSnapshot()
	})
})
