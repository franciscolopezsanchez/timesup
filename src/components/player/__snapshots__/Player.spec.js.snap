// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player tests renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="player"
  >
    <span>
      Jesus
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
